paddle/phi/kernels/funcs/algorithm.h

```diff
@@ -48,7 +48,7 @@ HOSTDEVICE inline size_t LowerBound(const T1 *x, size_t num, const T2 &val) {
   while (count > 0) {
     int64_t step = (count >> 1);
     auto *it = first + step;
-    if (*it < val) {
+    if (static_cast<float>(*it) < static_cast<float>(val)) {
```

Comment:
Sorry, the Chinese input method does not work in my OS temporarily.
Suggest that do not cast to 'float' directly, please use the MyTypeTrait in the document to infer the data type you need cast to.